wdl-lint/src/rules/blanks_between_elements.rs

```diff
@@ -0,0 +1,557 @@
+//! A lint rule for blank spacing between elements.
+
+use rowan::NodeOrToken;
+use wdl_ast::v1::InputSection;
+use wdl_ast::AstNode;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+use wdl_ast::SyntaxNode;
+use wdl_ast::SyntaxToken;
+use wdl_ast::ToSpan;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the blanks between elements rule.
+const ID: &str = "BlanksBetweenElements";
+
+/// Creates an excessive blank line diagnostic.
+fn excess_blank_line(span: Span) -> Diagnostic {
+    Diagnostic::note("extra blank line(s) found")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("remove blank line(s)")
+}
+
+/// Creates a missing blank line diagnostic.
+fn missing_blank_line(span: Span) -> Diagnostic {
+    Diagnostic::note("missing blank line")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("add blank line before this element")
+}
+
+/// Detects unsorted input declarations.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct BlanksBetweenElementsRule {
```

Comment:
I think this would be better implemented as an enum. Only one can be true at a time, right?